pkg/webhook/resources/setting/validator.go

```diff
@@ -1451,6 +1460,60 @@ func (v *settingValidator) checkVlanStatusReady(config *networkutil.Config) erro
 	return nil
 }
 
+func (v *settingValidator) checkLeftoverPodIPs(oldSetting, newSetting *v1beta1.Setting) error {
+	if oldSetting.Value == newSetting.Value {
+		return nil
+	}
+
+	if oldSetting.Value != "" && newSetting.Value == "" {
+		return nil
+	}
+
+	if oldSetting.Value != "" && newSetting.Value != "" {
+		return nil
+	}
+
+	pods, err := v.podCache.List(util.LonghornSystemNamespaceName, labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	podNames := filterPodsByStorageNetwork(pods)
+	if len(podNames) > 0 {
+		return fmt.Errorf("the following pods: %s still contain IPs from previous storage network; clean them up before re-enabling", strings.Join(podNames, ", "))
+	}
+
+	return nil
+}
+
+func filterPodsByStorageNetwork(pods []*corev1.Pod) []string {
+	podsFromStorageNetwork := make([]string, 0)
+	for _, pod := range pods {
+		networkInterface := pod.Annotations[string(lhmtypes.CNIAnnotationNetworkStatus)]
+		if networkInterface != "" && isPartOfStorageNetwork(networkInterface) {
+			podsFromStorageNetwork = append(podsFromStorageNetwork, pod.Name)
+			continue
```

Comment:
The original `continue` isn’t needed, or we could rewrite it as:

```suggestion
		networkInterface := pod.Annotations[string(lhmtypes.CNIAnnotationNetworkStatus)]
		if networkInterface == "" || !isPartOfStorageNetwork(networkInterface) {
			continue
		}
		podsFromStorageNetwork = append(podsFromStorageNetwork, pod.Name)
```

Reply:
You are right, originally the logic was diffident, but forgot to remove continue good catch

Reply:
Done :heavy_check_mark: